ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PostgresStorageBackend.scala

```diff
@@ -86,6 +88,37 @@ private[backend] object PostgresStorageBackend
 
   override val duplicateKeyError: String = "duplicate key"
 
+  def getPostgresVersion(connection: Connection): Option[(Int, Int, Int)] = {
+    val version = SQL"SHOW server_version".as(get[String](1).single)(connection)
+    val versionPattern = """(\d)[.](\d)[.](\d)""".r
+    version match {
+      case versionPattern(major, minor, patch) => Some(major.toInt, minor.toInt, patch.toInt)
+      case _ => None
+    }
+  }
+
+  override def checkCompatibility(
```

Comment:
since this feature seems to be only about logging, why not just add this to the main entrypoint of connection creation: DataSourceStorageBackend.createDataSource, and only where we need it?
This way we could avoid increasing StorageBackend and LedgerDao surface.

Reply:
The version is now checked within `createDataSource`.